api/agentpb/collector.proto

```diff
@@ -0,0 +1,325 @@
+syntax = "proto3";
+
+package agent;
+
+option go_package = "agentpb";
+
+import "inventorypb/agents.proto";
+import "qanpb/qan.proto";
+
+// MetricsBucket is aggregated message created by pmm-agent.
+// Contains information about one query selected in defined way from query class in specific period of time.
+message MetricsBucket {
+  // Common contains common fields for all DBs.
+  message Common {
+    // md5 of digest_text/fingerprint.
+    string queryid = 1;
+    // digest_text - query signature. Query without values.
+    string fingerprint = 2;
+    //
+    // Dimension Group.
+    //
+    // Service Name (can be several Services on one Node).
+    string service_name = 3;
+    // PostgreSQL database.
+    string database = 4;
+    // MySQL database or PostgreSQL schema.
+    string schema = 5;
+    // client user name.
+    string username = 6;
+    // client IP or hostname.
+    string client_host = 7;
+    //
+    // Unified Labels Group.
+    //
+    // Name of replication set
+    string replication_set = 10;
+    // Cluster name
+    string cluster = 11;
+    // Type of service
+    string service_type = 12;
+    // Environment name
+    string environment = 13;
+    // Availability zone
+    string az = 14;
+    // Region name
+    string region = 15;
+    // Node model
+    string node_model = 16;
+    // Container name
+    string container_name = 17;
+    // Identifier of agent that collect and send metrics.
+    string agent_id = 18;
+    // Metrics source.
+    inventory.AgentType agent_type = 19;
+    // Custom labels names:values.
+    map<string, string> labels = 20;
+    //
+    // Period and Query Example Group.
+    //
+
+    // Time when collection of bucket started.
+    uint32 period_start_unix_secs = 30;
+    // Duration of bucket.
+    uint32 period_length_secs = 31;
+    // One of query example from set found in bucket.
+    string example = 32;
+    qan.ExampleFormat example_format = 33;
+    // Indicates if query examples is too long and was truncated.
+    bool is_truncated = 34;
+    qan.ExampleType example_type = 35;
+    // Metrics of query example in JSON format.
+    string example_metrics = 36;
+  }
+  // MySQL contains fields only for MySQL.
+  message MySQL {
+    //
+    // Metrics.
+    //
+
+    // How many queries was with warnings in bucket.
+    float num_queries_with_warnings = 1;
+    // List of warnings: {code: count}.
+    map<uint64, uint64> warnings = 2;
+    // How many queries was with error in bucket.
+    float num_queries_with_errors = 3;
+    // List of errors: {code: count}.
+    map<uint64, uint64> errors = 4;
+    // Amount queries in this bucket.
+    float num_queries = 5;
+    // How many times query_time was found.
+    float m_query_time_cnt = 6;
+    // The statement execution time in seconds.
+    float m_query_time_sum = 7;
```

Comment:
I think m_query_time_sum - should be in common message as it is also valid for mongo and postgres.